libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1035,6 +1027,18 @@ void PbftManager::secondFinish_() {
   auto end_time_for_step = (2 + step_ - startingStepInRound_) * LAMBDA_ms - POLLING_INTERVAL_ms;
 
   updateSoftVotedBlockForThisRound_();
+  if (soft_voted_block_for_this_round_.first != NULL_BLOCK_HASH && soft_voted_block_for_this_round_.second) {
```

Comment:
And we can call this every second finishing step without re-gossiping, correct?

Reply:
Yes

Reply:
@justinsnapp I will resolve this conversation to be able to test the fix on devnet to get it ustuck. If there is something still unclear please follow up